packages/lodestar-validator/src/validator.ts

```diff
@@ -73,6 +75,54 @@ export class Validator {
     if (this.blockService) await this.blockService.stop();
   }
 
+  /**
+   * Perform a voluntary exit for the given validator by its public key.
+   */
+  public async voluntaryExit(publicKey: string): Promise<void> {
+    await this.apiClient.connect();
+
+    const stateValidator = await this.apiClient.beacon.state.getStateValidator(
+      "head",
+      this.config.types.BLSPubkey.fromJson(publicKey)
+    );
+    if (!stateValidator) throw new Error("No validator found in validator store.");
+
+    const epoch = this.apiClient.clock.currentEpoch;
+
+    const voluntaryExit = {
+      epoch,
+      validatorIndex: stateValidator.index,
+    };
+
+    const state: BeaconState = this.config.types.BeaconState.tree.defaultValue();
+    state.fork = (await this.apiClient.beacon.state.getFork("head")) || state.fork;
```

Comment:
you should fetch https://ethereum.github.io/eth2.0-APIs/#/Config/getForkSchedule and pick fork version based on voluntary exit epoch (if you wanna schedule exit in the future you need to use future fork version). I think it's fine to use (fallback) to (this.apiClient.beacon.state.getFork("head")) if nothing is scheduled

Reply:
working on this api implementation here: #2036